PIVX-Labs/MyPIVXWallet
# Resync from scratch if sapling root different from what's advertised

## Abstract

Resync from scratch if the sapling root is different from what advertised in the explorer.
A few different things may cause this:
- A bug in the syncing process
- rpc node skipping some shield blocks
- Explorer forking

A few users have reported this happening, while it is fixable by simply reimporting the wallet, this should do it automatically.

## Testing
- Switch to mainnet (Testnet explorer doesn't give the sapling root)
- Get an invalid wallet sapling state or replace sapling root check with something like `Math.random() > 0.5`
-  Check that MPW tries to resync automatically
